src/kaltura-player.js

```diff
@@ -53,16 +55,50 @@ class KalturaPlayer extends FakeEventTarget {
   _sourceSelected: boolean = false;
   _pluginReadinessMiddleware: PluginReadinessMiddleware;
   _configEvaluator: ConfigEvaluator;
+  _viewabilityManager: ViewabilityManager;
+  _visibilityTabChangeEventName: string;
+  _visibilityTabHiddenAttr: string;
+  _playbackStart: boolean;
+
+  /**
+   * Whether the player is visible in the page scroll view
+   * @type {boolean}
+   * @private
+   */
+  _isVisibleInScroll: boolean;
+
+  /**
+   * Whether the player browser tab is active or not
+   * @type {boolean}
+   * @private
+   */
+  _isTabVisible: boolean;
+
+  /**
+   * Whether the player browser tab is active and in the scroll view
+   * @type {boolean}
+   * @private
+   */
+  _isVisible: boolean;
+
+  /**
+   * Whether the player was auto paused
+   * @type {boolean}
+   * @private
+   */
+  _autoPaused: boolean = false;
 
   constructor(options: KPOptionsObject) {
     super();
     const {sources, plugins} = options;
     this._configEvaluator = new ConfigEvaluator();
+    this._playbackStart = false;
     this._configEvaluator.evaluatePluginsConfig(plugins, options);
     const noSourcesOptions = Utils.Object.mergeDeep({}, options, {sources: null});
     delete noSourcesOptions.plugins;
     this._localPlayer = loadPlayer(noSourcesOptions);
     this._controllerProvider = new ControllerProvider(this._pluginManager);
+    this._viewabilityManager = new ViewabilityManager(this.config.visibility.tolerance);
```

Comment:
why not pass entire visibility config to manager?

Reply:
done